backend/middlewares/checkAuth.js

```diff
@@ -0,0 +1,14 @@
+const jwt = require("jsonwebtoken");
+
+module.exports = (req, res, next) => {
+  try {
+    const token = req.headers["x-auth-token"];
+    // jwt verify also throws error on failure
+    // hence in try catch block
+    const decodedToken = jwt.verify(token, process.env.SECRET_KEY);
+    req.user = decodedToken.id;

```

Comment:
Are you sure, decodedToken.id will return user object, not just email or any id? Because in next file you are using req.user.email...

Reply:
While encoding the token in backend/utils/token.utils.js the id field has been set as the user's email, so we use the same name after decoding the token